### DISCUSSION TITLE
New custom metrics

### DISCUSSION THREAD

**Discussion by aakashm07** [cybertec-postgresql/pgwatch2#657] [Q&A]
Hi Team,

    I am trying to add custom metrics like below Under metric definition page

Name: active_connection_per_client_address
Query: SELECT (extract(epoch from now()) * 1e9)::int8 as epoch_ns,client_addr,count(*) from pg_stat_activity where state='active' group by 2 ;
is active - yes
is helper - yes
metrics atrributes- {"is_instance_level": true}
column attributes - {"prometheus_all_gauge_columns": true}

Added these metrics under custom config "active_connection_per_client_address": 60} 

Even tried to add it under full and mapped the DB into full.

But now the question is it is not showing any data. In influx(data store), there is no tables created for this. We can create custom dashboards over it, but there is no metrics stored. Should we need to do anything on data store ?

Also for sequences, Data is there in influx, but there is no dashboard to present data of it. Kindly help here. Thanks

**Comment by pashagolub** [comment]
Hello.

> 

> Hi Team,
> 
> ```
> I am trying to add custom metrics like below Under metric definition page
> ```
> 
> Name: active_connection_per_client_address Query: SELECT (extract(epoch from now()) * 1e9)::int8 as epoch_ns,client_addr,count(*) from pg_stat_activity where state='active' group by 2 ; is active - yes is helper - yes 

Helper is a special function used in the metric SQL, e.g. `get_cpu_load()`. What you did is defined this SQL **not as a metric but as a helper function**. That's why you don't see anything. The system waits how to use this new helper but never meets it's usage.

> metrics atrributes- {"is_instance_level": true} column attributes - {"prometheus_all_gauge_columns": true}
> 
> Added these metrics under custom config "active_connection_per_client_address": 60}
> 
> Even tried to add it under full and mapped the DB into full.
> 
> But now the question is it is not showing any data. In influx(data store), there is no tables created for this. We can create custom dashboards over it, but there is no metrics stored. Should we need to do anything on data store ?
> 
> Also for sequences, Data is there in influx, but there is no dashboard to present data of it. Kindly help here. Thanks

Just as a note. Influx is a deprecated data storage. You should move to PostgreSQL or even better to TimescaleDB
